activation/nipost.go

```diff
@@ -479,3 +495,21 @@ func constructMerkleProof(challenge types.Hash32, members []types.Member) (*type
 		Nodes:     nodesH32,
 	}, nil
 }
+
+func randomDurationInRange(min, max time.Duration) time.Duration {
+	return min + time.Duration(float64((max-min).Nanoseconds())*rand.Float64())
```

Comment:
```suggestion
	return min + time.Duration(rand.Int63n(int64(max-min)))
```

should be equivalent but avoids conversions to and from `float64` (`time.Duration` is already an `int64` so the cast should be free).